codec-http2/src/main/java/io/netty5/handler/codec/http2/Http2Exception.java

```diff
@@ -29,7 +29,7 @@
  * Exception thrown when an HTTP/2 error was encountered.
  */
 @UnstableApi
-public class Http2Exception extends Exception {
+public class Http2Exception extends RuntimeException {
```

Comment:
This change seems to be unrelated and maybe not what we want

Reply:
Yeah, this change is because `DefaultHttp2Headers.validateKeys` do not declare any checked exceptions. Previously we used a sneaky-throw to bypass the compile-time check for this. Making this a `RuntimeException` is a pretty blunt hammer.
The alternative would be to throw `IllegalArgumentException` (or a new class derived from it) like the `MultiMap` derived headers used to do, and then catch that in all the places where we build header objects.